app/controllers/application_controller/buttons.rb

```diff
@@ -346,12 +346,7 @@ def group_button_add_save(typ)
     end
     group_set_record_vars(@custom_button_set)
 
-    mems = []
-    @edit[:new][:fields].each_with_index do |field,idx|
-      uri = CustomButton.find(field[1])
-      uri.save
-      mems.push(uri)
-    end
+    mems = @edit[:new][:fields].collect { |field| CustomButton.find(field[1]) }
```

Comment:
@Fryguy - I saw the N+1 and did not want to chase a shiny object during refactoring.  I did remove the `uri.save` immediately after finding it from the database.  ;-)  Now that I can see the forest for the trees here, will refactor to remove the N+1.